main.sagews

```diff
@@ -28,4 +28,49 @@ for i in range(m):
     for j in range(n):
             PK[i,j] = a[i,j]
     PK[i,n] = Mod(Mod(a[i].inner_product(s),p)+e[i],p)
-print "Public Key is: ",PK
\ No newline at end of file
+print "Public Key is: ",PK
+
+#ENCRYPTION--------------------------------------------
+
+#Random size of random set S
+Ssize=int(random()*(m+1))
+print "The number of elements of random set S is: ",Ssize
+
+#Random Set S initilization
+S=matrix(Ssize,n+1)
+
+#This list counts the times a random row of Public Key is chosen to be entered in S.
+counter=[0 for i in range(m)]
+
+#Random rows of Public Key are entered in S.
+for i in range(Ssize):
+    rand=int(random()*(m))
+    counter[rand]+=1
+    #No duplicate rows
+    if counter[rand]<2:
+    	for l in range(n+1):
+    	    S[i,l]=PK[rand,l]
+print "Random Set is: ",S
+
+#A message in binary
+Message=[0,1,1,0,1,0,1,1,0,0]
+Messagelength=10
+
+#EncryptedMessage matrix initialization
+EncryptedMessage=matrix(Messagelength,2)
+
+#Encryption Proccess
+for i in range(Messagelength):
```

Comment:
Για κάθε bit του μηνύματος, πρέπει να αλλάζει το S. Επομένως ότι έκανες πιο πάνω για το S, θα το βάλεις όλο εδώ μέσα. Η κάνε απλά μια συνάρτηση που να έχει όλον αυτόν τον κώδικα για το S και να επιστρέφει απλά το S, με return.
def createS():
    κώδικας για το S
    return S

Μέσα στο loop for i in range(Messagelength):
S = createS()